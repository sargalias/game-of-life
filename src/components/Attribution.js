import React from 'react';

const Attribution = (props) => (
  <div id="attribution">
    <p id="author">
      By <a href={props.authorUrl} target="_blank">{props.authorName}</a>
    </p>
    <p id="code-repository">
      View code on <a href={props.codeRepositoryUrl} target="_blank">GitHub</a>
    </p>
  </div>
);

export default Attribution;