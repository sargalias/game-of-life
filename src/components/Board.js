import React from 'react';
import Row from './Row';
import generateGradient from "../logic/color";
import { generateEmptyBoard } from "../logic/board";

class Board extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      colorCount: 0
    };
    const {hasColor} = props;
    if (hasColor) {
      this.state.colors = generateGradient();
    }
  }

  _gen2DColorArray = () => {
    const { boardData, generation=0 } = this.props;
    const { colors } = this.state;
    const rows = boardData.length;
    const columns = rows >= 1 ? boardData[0].length : 0;
    const colorBoard = generateEmptyBoard(rows, columns);
    for (let row=0; row<rows; row++) {
      for (let column=0; column<columns; column++) {
        const colorIndex = (column + row + generation) % colors.length;
        colorBoard[row][column] = colors[colorIndex];
      }
    }
    return colorBoard;
  };

  render() {
    let colorBoard = this.props.hasColor ? this._gen2DColorArray() : undefined;
    return (
      <div className="board">
        {this.props.boardData.map((rowData, index) => (
          <Row
            key={index}
            rowId={index}
            rowData={rowData}
            rowColorData={colorBoard && colorBoard[index]}
            onCellClick={this.props.onCellClick}
          />
        ))}
      </div>
    );
  }
}

export default Board;
