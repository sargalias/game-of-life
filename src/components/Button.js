import React from 'react';

const Button = (props) => (
  <button className="button">{props.text}</button>
);

export default Button;
