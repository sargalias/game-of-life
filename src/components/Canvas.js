import React from 'react';

const Canvas = () => (
  <div id="canvas-container">
    <canvas id="canvas" />
  </div>
);

export default Canvas;
