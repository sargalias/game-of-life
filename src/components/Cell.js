import React from 'react';

const Cell = (props) => (
  <div
    className={`cell ${props.value === 0 ? 'dead' : 'alive'}`}
  ></div>
);

export default Cell;
