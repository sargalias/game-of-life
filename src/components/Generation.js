import React from 'react';

const Generation = () => (
  <div id="generation">
    Generation: <span id="generation-number">0</span>
  </div>
);

export default Generation;
