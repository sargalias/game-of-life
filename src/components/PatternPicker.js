import React from 'react';

const PatternPicker = ({patterns=[], onChange, value}) => (
  <div className="pattern-picker-container">
    <div>
      <label htmlFor="pattern" className="pattern-picker-text">Pattern:</label>
    </div>
    <select name="pattern" id="pattern" onChange={onChange} value={value}>
      {patterns.map((pattern) => (
        <option value={pattern} key={pattern}>{pattern}</option>))
      }
    </select>
  </div>
);

export default PatternPicker;
