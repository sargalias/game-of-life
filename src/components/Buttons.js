import React from 'react'
import Button from './Button';

const Buttons = () => (
  <div className="button-container">
    <Button text="Run" />
    <Button text="Stop" />
    <Button text="Clear" />
    <Button text="Reset" />
  </div>
);

export default Buttons;
