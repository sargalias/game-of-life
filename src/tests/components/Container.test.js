import React from 'react';
import { shallow } from 'enzyme';
import Container from '../../components/Container';
import { PlainComponent, ClassnameComponent } from '../fixtures/components';

test('should render container correctly with child components', () => {
  const wrapper = shallow(<Container>
    <PlainComponent/>
    <ClassnameComponent/>
  </Container>);
  expect(wrapper.children().length).toBe(2);
  expect(wrapper.contains([<PlainComponent/>, <ClassnameComponent/>]));
  expect(wrapper).toMatchSnapshot();
});
