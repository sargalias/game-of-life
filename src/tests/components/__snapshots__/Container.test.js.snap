// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render container correctly with child components 1`] = `
<div
  className="container"
>
  <PlainComponent />
  <ClassnameComponent />
</div>
`;
